build.gradle

```diff
@@ -1,284 +1,149 @@
-          args << '-Wno-psabi' << '-Wno-unused-parameter' << '-fPIC' << '-O0' << '-g3' << '-rdynamic'
-          //TODO: When the compiler allows us to actually call deprecated functions from within
-          // deprecated function, remove this line (this will cause calling deprecated functions
-          // to be treated as a warning rather than an error).
-          args << '-Wno-error=deprecated-declarations'
-          args.remove('-m32')
-        }
-        linker.withArguments { args ->
-          args << '-rdynamic'
-          args.remove('-m32')
-        }
-        staticLibArchiver.executable = compilerPrefix + staticLibArchiver.executable
-      }
-      target('x86') {
-        cppCompiler.withArguments { args ->
-          args << '-std=c++11' << '-Wformat=2' << '-Wall' << '-Wextra' << '-Werror' << '-pedantic'
-          args << '-Wno-psabi' << '-Wno-unused-parameter' << '-fPIC' << '-O0' << '-g3' << '-rdynamic'
-          //TODO: When the compiler allows us to actually call deprecated functions from within
-          // deprecated function, remove this line (this will cause calling deprecated functions
-          // to be treated as a warning rather than an error).
-          args << '-Wno-error=deprecated-declarations'
-          args << '-m32'
-        }
-        linker.withArguments { args ->
-          args << '-rdynamic'
-          args << '-m32'
-        }
-      }
-      target('x64') {
-        cppCompiler.withArguments { args ->
-          args << '-std=c++11' << '-Wformat=2' << '-Wall' << '-Wextra' << '-Werror' << '-pedantic'
-          args << '-Wno-psabi' << '-Wno-unused-parameter' << '-fPIC' << '-O0' << '-g3' << '-rdynamic'
-          //TODO: When the compiler allows us to actually call deprecated functions from within
-          // deprecated function, remove this line (this will cause calling deprecated functions
-          // to be treated as a warning rather than an error).
-          args << '-Wno-error=deprecated-declarations'
-        }
-        linker.withArguments { args ->
-          args << '-rdynamic'
-        }
-      }
-    }
-    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-      visualCpp(VisualCpp) {
-        def vs14Dir = file('C:\\Program Files (x86)\\Microsoft Visual Studio 14.0')
-        // If vs2015 is installed, fall back to vs2013 for now, until Gradle pulls in the right
-        // includes for the win10 sdk.
-        if (vs14Dir.exists()) {
-          def vs12Dir = file('C:\\Program Files (x86)\\Microsoft Visual Studio 12.0')
-          if (!vs12Dir.exists()) {
-            throw new GradleException('Compilation with VS 2015 is not currently supported by Gradle. Please install VS 2013, or use the CMake build option.')
-          }
-          installDir = vs12Dir
+ext.getPlatformPath = { binary ->
+    if (binary.targetPlatform.architecture.arm) {
+        return 'Linux/arm'
+    } else if (binary.targetPlatform.operatingSystem.name == 'Linux') {
```

Comment:
On my system, this needs to be if (binary.targetPlatform.operatingSystem.linux) and similarly I think the Windows line needs to be if (binary.targetPlatform.operatingSystem.windows).